doc/source/whatsnew/v1.5.0.rst

```diff
@@ -267,6 +267,7 @@ Categorical
 ^^^^^^^^^^^
 - Bug in :meth:`Categorical.view` not accepting integer dtypes (:issue:`25464`)
 - Bug in :meth:`CategoricalIndex.union` when the index's categories are integer-dtype and the index contains ``NaN`` values incorrectly raising instead of casting to ``float64`` (:issue:`45362`)
+- Bug in :meth:`CategoricalIndex._concat` when concatenating categorical indexes that have the same categories returning indexes in improper order (:issue:`44099`)
```

Comment:
can you refer to `pd.concat` as that's the most user visible bug here

Reply:
Hi, thank you for review!

I added description in this commit(https://github.com/pandas-dev/pandas/pull/46019/commits/aacc75c6807b882f6142007f8802d92f80815a74)